app/models/effect.rb

```diff
@@ -17,4 +19,47 @@ class Effect < ApplicationRecord
     **PERSONAL_EFFECTS,
     **MAGICAL_EFFECTS
   }
+
+
+  def self.effect_sub_obj
+    effects_subs =  {visual: VISUAL_EFFECTS,
+          special: SPECIAL_EFFECTS,
+          personal: PERSONAL_EFFECTS,
+          magical: MAGICAL_EFFECTS}
+    main_obj = {}
+
+    for effect in effects_subs.keys
+        sub_types = []
+
+        for key in effects_subs[effect].keys
+            sub_types << key
+        end
+
+        main_obj[effect] = sub_types
+    end
+    main_obj
+  end
```

Comment:
Good use of the class method. Only comment is that you ended up going to quite a lot of trouble to create you effect_sub_object, but you could have done it quite quickly with:
```
main_obj = {
  visual: VISUAL_EFFECTS.keys,
  special: SPECIAL_EFFECTS.keys,
  ...
}
```
If you wanted to be even lazier about it, you could use some meta-programming...
```
effect_types.each_with_object({}) do |effect, memo|
  memo[effect] = "#{effect.upcase}_EFFECTS".constantize.keys
end
```
NB. I never would have expected you to do this given that you've only just started looking at Ruby. Mostly just showing you for interest.